### PR TITLE
Use clamp to edge on vulkan and enable portability features

### LINKED ISSUE #162: vulkan: Samplers should be "clamp to edge", not "repeat"
yakui's wgpu backend uses "clamp to edge" samplers for all images to prevent blending artifacts. The Vulkan backend currently always uses "repeat" images, which can cause unique blending issues when images are resized.

### LINKED ISSUE #163: vulkan: Demo is not portability compliant
## Background
As per the [MoltenVK docs](https://github.com/KhronosGroup/MoltenVK?tab=readme-ov-file#using-the-vulkan-sdk), when using a backend that supports the `VK_KHR_portability_subset` (ie. MoltenVK), we must enable the correct extensions.

### PR DESCRIPTION
**Pull request by kanerogers**
Some small Vulkan fixes.

- Closes #162 
- Closes #163

### CODE REVIEW

File: crates/yakui-vulkan/src/vulkan_texture.rs
Comment (LPGhatguy) [inline crates/yakui-vulkan/src/vulkan_texture.rs:62]: Does the other mention of `REPEAT` in this file also need to be updated?

File: crates/yakui-vulkan/examples/demo.rs
Comment (LPGhatguy) [inline crates/yakui-vulkan/examples/demo.rs:289]: What's the impact of checking for macOS and iOS versus just iOS? Do they support different extensions?

File: crates/yakui-vulkan/examples/demo.rs
Comment (LPGhatguy) [inline crates/yakui-vulkan/examples/demo.rs:355]:
```suggestion
        #[cfg(target_os = "macos")]
```

**Review by LPGhatguy — COMMENTED** [review]
A couple comments and questions from me, but looks good!

File: crates/yakui-vulkan/examples/demo.rs
Comment (kanerogers) [reply crates/yakui-vulkan/examples/demo.rs:289]: Just a bad copypasta, sorry. Since we don't support iOS anyway I'll clean this up to just be macOS.

File: crates/yakui-vulkan/src/vulkan_texture.rs
Comment (kanerogers) [reply crates/yakui-vulkan/src/vulkan_texture.rs:62]: Agh, yes.

**Review by LPGhatguy — APPROVED** [review]
LGTM! Just waiting on Ralith's sign-off.

File: crates/yakui-vulkan/src/vulkan_texture.rs
Comment (Ralith) [inline crates/yakui-vulkan/src/vulkan_texture.rs:103]: Shouldn't this be in the previous commit?

**Review by Ralith — APPROVED** [review]

**Pull request merged**
